This patch adds mapfile option to libdns_sd.so.1 link command. We need that in
order to keep compatibility with the previous mDNS version.

--- a/avahi-compat-libdns_sd/Makefile.am
+++ b/avahi-compat-libdns_sd/Makefile.am
@@ -42,7 +42,7 @@ txt.c \
 	unsupported.c \
 	warn.c warn.h
 libdns_sd_la_CFLAGS = $(AM_CFLAGS) $(PTHREAD_CFLAGS)
-libdns_sd_la_LDFLAGS = $(AM_LDFLAGS)  -version-info $(LIBAVAHI_COMPAT_LIBDNS_SD_VERSION_INFO) $(PTHREAD_LIBS) $(PTHREAD_CFLAGS) ../avahi-common/libavahi-common.la ../avahi-client/libavahi-client.la
+libdns_sd_la_LDFLAGS = $(AM_LDFLAGS) -Wl,-M,../../../mapfile -version-info $(LIBAVAHI_COMPAT_LIBDNS_SD_VERSION_INFO) $(PTHREAD_LIBS) $(PTHREAD_CFLAGS) ../avahi-common/libavahi-common.la ../avahi-client/libavahi-client.la
 
 txt_test_SOURCES = \
 	dns_sd.h \
